### PR TITLE
[New Rule] Shared Object Created

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This PR proposes the addition of a detection rule that monitors the creation and change of `shared object files` within the `/usr/lib` or `/dev/shm` directories. 

The creation of a shared object file involves compiling code into a dynamically linked library that can be loaded by 
other programs at runtime. While this process is typically used for legitimate purposes, malicious actors can leverage
shared object files (e.g., .so files) to execute unauthorized code, inject malicious functionality into legitimate
processes, or bypass security controls. This allows malware to persist on the system, evade detection, and potentially
compromise the integrity and confidentiality of the affected system and its data. This rule monitors the creation of
shared object files by previously unknown processes. 

The rule has been checked in clusters with several Linux servers with data ranging back over half a year ago, and does not trigger any false positives based on the exclusions added such as `dockerd`. Additionally, it detected most if not all malicious shared object file creations originating from the Linux malware. Additionally, a new terms rule was chosen in order to minimize potential FPs even further. 

## Detection
```
host.os.type : "linux" and event.action:("creation" or "file_create_event" or "rename" or "file_rename_event") and 
file.path : (/usr/lib/* or /dev/shm/*) and file.extension : "so" and process.name : * and not 
process.name : ("dpkg" or "dockerd" or "rpm" or "snapd" or "5")

[rule.new_terms]
field = "new_terms_fields"
value = ["file.path", "process.name"]

[[rule.new_terms.history_window_start]]
field = "history_window_start"
value = "now-7d"
```
image

## Malware Analysis
These rules are found to be triggered by real malware samples, which @1337-42 and I have been detonating and analysing. Updates will be added to [#105](https://github.com/elastic/elastic-security-labs/issues/105).

### CODE REVIEW

File: rules/linux/persistence_shared_object_creation.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+The creation of a shared object file involves compiling code into a dynamically linked library that can be loaded by 
+other programs at runtime. While this process is typically used for legitimate purposes, malicious actors can leverage
+shared object files (e.g., .so files) to execute unauthorized code, inject malicious functionality into legitimate
+processes, or bypass security controls. This allows malware to persist on the system, evade detection, and potentially
+compromise the integrity and confidentiality of the affected system and its data. This rule monitors the creation of
+shared object files by previously unknown processes.
```
Comment (imays11) [inline rules/linux/persistence_shared_object_creation.toml:17]:
```suggestion
This rule monitors the creation ofshared object files by previously unknown processes. 
The creation of a shared object file involves compiling code into a dynamically linked library that can be loaded by 
other programs at runtime. While this process is typically used for legitimate purposes, malicious actors can leverage
shared object files (e.g., .so files) to execute unauthorized code, inject malicious functionality into legitimate
processes, or bypass security controls. This allows malware to persist on the system, evade detection, and potentially
compromise the integrity and confidentiality of the affected system and its data. 
```
maybe consider moving your last line to the front.

File: rules/linux/persistence_shared_object_creation.toml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+name = "Persistence"
+reference = "https://attack.mitre.org/tactics/TA0003/"
+
+[rule.new_terms]
+field = "new_terms_fields"
+value = ["file.path", "process.name"]
+
+[[rule.new_terms.history_window_start]]
+field = "history_window_start"
+value = "now-7d"
```
Comment (imays11) [inline rules/linux/persistence_shared_object_creation.toml:58]: great use of new_terms, why such a short window?

**Review by imays11 — APPROVED** [review]

File: rules/linux/persistence_shared_object_creation.toml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+updated_date = "2023/06/09"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule monitors the creation of shared object files by previously unknown processes. The creation of a shared object 
+file involves compiling code into a dynamically linked library that can be loaded by other programs at runtime. While 
+this process is typically used for legitimate purposes, malicious actors can leverage shared object files to execute 
+unauthorized code, inject malicious functionality into legitimate processes, or bypass security controls. This allows 
+malware to persist on the system, evade detection, and potentiallycompromise the integrity and confidentiality of the
```
Comment (w0rk3r) [inline rules/linux/persistence_shared_object_creation.toml:16]:
```suggestion
malware to persist on the system, evade detection, and potentially compromise the integrity and confidentiality of the 
```

File: rules/linux/persistence_shared_object_creation.toml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+name = "Persistence"
+reference = "https://attack.mitre.org/tactics/TA0003/"
+
+[rule.new_terms]
+field = "new_terms_fields"
+value = ["file.path", "process.name"]
+
+[[rule.new_terms.history_window_start]]
+field = "history_window_start"
+value = "now-7d"
```
Comment (w0rk3r) [inline rules/linux/persistence_shared_object_creation.toml:58]:
```suggestion
value = "now-7d"

```

**Review by w0rk3r — APPROVED** [review]
LGTM

File: rules/linux/persistence_shared_object_creation.toml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+file.path : (/usr/lib/* or /dev/shm/*) and file.extension : "so" and process.name : * and not 
+process.name : ("dpkg" or "dockerd" or "rpm" or "snapd" or "5")
+'''
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1574"
+name = "Hijack Execution Flow"
+reference = "https://attack.mitre.org/techniques/T1574/"
```
[the remark this answers is not in the document]
Comment (Aegrah) [reply rules/linux/persistence_shared_object_creation.toml:45]: I agree, added it, thx!

**Pull request merged**